Source/RapyutaSimulationPlugins/Public/Robots/RRRobotStructs.h

```diff
@@ -1423,12 +1463,209 @@ struct RAPYUTASIMULATIONPLUGINS_API FRRRobotModelInfo : public TSharedFromThis<F
         UE_LOG_WITH_INFO(LogTemp, Display, TEXT("WholeBodyStaticMeshName: %s"), *WholeBodyStaticMeshName);
         WholeBodyMaterialInfo.PrintSelf();
 
-        for (const auto& modelInfo : ChildModelsInfo)
+        for (const auto& modelInfo : ChildModelsData)
         {
             UE_LOG_WITH_INFO(LogTemp, Warning, TEXT("SUBMODEL"));
             modelInfo.PrintSelf();
         }
     }
+};    // END FRRRobotModelData
+
+USTRUCT(BlueprintType)
```

Comment:
What is the main motivation to introduce these struct with almost just accessor?

Reply:
As noted in the description, basically we need `FRRRobotModelInfo` as the ultimate in-memory robot model data, either loaded from URDF or from UDataTable.

They are meant for different purposes and thus different parent structs:
* `FRRRobotModelInfo : public TSharedFromThis<FRRRobotModelInfo, ESPMode::ThreadSafe>` -> to allow access as `TSharedRef` & thread-safe, since robot model info could be accessed from any thread.
* `FRRRobotModelTableRow : public FTableRowBase` -> only meant to store data in data table for BP robot users.

Both of these thus need to wrap the core robot model data stored in `FRRRobotModelData`.

Reply:
As noted in the description, basically we need `FRRRobotModelInfo` as the ultimate in-memory robot model data, either loaded from URDF or from UDataTable.

They are meant for different purposes and thus different parent structs:
* `FRRRobotModelInfo : public TSharedFromThis<FRRRobotModelInfo, ESPMode::ThreadSafe>` -> to allow access as `TSharedRef` & thread-safe, since robot model info could be accessed from any thread. This does not support USTRUCT() thus cannot be used directly for storing data in UDataTable.
* `FRRRobotModelTableRow : public FTableRowBase` -> only meant to store data in data table for BP robot users.

Both of these thus need to wrap the core robot model data stored in `FRRRobotModelData`.

Reply:
For my understainding
- FRRRobotModelInfo provide Thread safe getter for FRRRobotModelData
- Does UDataTalbe require child struct of FTableRowBase only with data?

Reply:
Yeah it does
https://github.com/EpicGames/UnrealEngine/blob/release/Engine/Source/Runtime/Engine/Classes/Engine/DataTable.h#L315
As the name `FRRRobotModelTableRow` itself should already denote its meaning, I'll add some docs